LiPTrPpipeline.R

```diff
@@ -340,224 +549,418 @@ for (i in seq_along(comparisons)) {
-    rowwise() %>% 
-    dplyr::mutate(gene = sort(strsplit(gene_names, " ", fixed = TRUE)[[1]])[1]) %>% 
-    ungroup() %>% 
-    dplyr::mutate(label = paste(gene, "@", start, "-", end, sep = "")) %>% 
-    dplyr::mutate(significant = TRUE)
-    
-  print(paste("Number of significantly changing candidates: ", length(unique(candidates$eg_modified_peptide)), " peptide(s) belonging to ", length(unique(candidates$pg_protein_accessions)), " protein(s)."), sep = "")
-  print(distinct(candidates, pg_protein_accessions, eg_modified_peptide, label))
-  
-  df_diff_abundance <- df_diff_abundance %>%
-    left_join(distinct(candidates, eg_modified_peptide, label, significant), by = "eg_modified_peptide") %>% 
-    left_join(dplyr::distinct(DIA_clean_uniprot, pg_protein_accessions, gene_names, sequence, length, coverage), by = "pg_protein_accessions") %>% 
-    rowwise() %>% 
-    dplyr::mutate(gene = sort(strsplit(gene_names, " ", fixed = TRUE)[[1]])[1]) 
-  
-  # Save Differential Abundance results
-  diff_abundance_file <- file.path(
-    group_folder_path, 
-    paste0("differential_abundance_", experiment_id, "_", comparison_filter, ".csv")
-  )
+  
+  if (!is.null(input_file_tryptic_control)) {
+    diff_trp <- tryptic_protein_abundance %>%
+      assign_missingness(
+        sample = r_file_name,
+        condition = r_condition,
+        intensity = normalised_intensity_log2,
+        grouping = pg_protein_accessions,
+        ref_condition = "CTR_Trp",
+        retain_columns = all_of(c("pg_protein_accessions2"))
+      ) %>%
+      calculate_diff_abundance(
+        sample = r_file_name,
+        condition = r_condition,
+        grouping = pg_protein_accessions,
+        intensity_log2 = normalised_intensity_log2,
+        comparison = comparison,
+        method = "t-test",
+        retain_columns = all_of(c("pg_protein_accessions2"))
+      )
+    
+    Trp_candidates <- diff_trp %>% 
+      dplyr::filter(adj_pval < 0.05 & abs(diff) > 1) %>% 
+      left_join(dplyr::distinct(tryptic_protein_abundance, pg_protein_accessions2, gene_names), by = "pg_protein_accessions2") %>% 
+      rowwise() %>% 
+      dplyr::mutate(gene = sort(strsplit(gene_names, " ", fixed = TRUE)[[1]])[1]) %>% 
+      ungroup() %>% 
+      dplyr::mutate(significant = TRUE)
+    
+    print(paste("Tryptic control: Number of significantly changing candidates: ", length(unique(Trp_candidates$pg_protein_accessions)), " protein(s)."), sep = "")
+    print(distinct(Trp_candidates, gene))
+    
+    diff_trp <- diff_trp %>%
+      left_join(distinct(Trp_candidates, pg_protein_accessions2, significant, gene), by = "pg_protein_accessions2") %>% 
+      left_join(dplyr::distinct(tryptic_protein_abundance, pg_protein_accessions2, gene_names), by = "pg_protein_accessions2")
+
+    
+    # BUG: did the yaml parameter changed? 
+    # before: experiment_id => experiment_ids
```

Comment:
the experiment ids are used in a for loop
for (i in seq_along(comparisons)) {
  comparison_filter <- comparisons[[i]]
  experiment_id <- experiment_ids[[i]]